src/integration-test/java/com/commercetools/sync/integration/ctpprojectsource/products/ProductSyncIT.java

```diff
@@ -409,7 +445,8 @@ void sync_withProductTypeReference_ShouldUpdateProducts() {
 
         assertThat(updateActions).containsExactlyInAnyOrder(
             SetAttributeInAllVariants.of(targetProductRefAttr.getName(), targetProductRefAttr.getValue(), true),
-            SetAttributeInAllVariants.of(targetProductSetRefAttr.getName(), targetProductSetRefAttr.getValue(), true)
+            SetAttributeInAllVariants.of(targetProductSetRefAttr.getName(), targetProductSetRefAttr.getValue(), true),
+            Publish.of()
```

Comment:
What happened here in this test that we got additionally published action?

Reply:
Some tests are adjusted after the fix, this case was missing before. I did not change the current prepared products.

In this case, when the target product is published a product is updated, the product will `hasStagedChanges=true`, stage and current will be different and sync needs to send a publish to make sure changes are synced between staged and current projections.